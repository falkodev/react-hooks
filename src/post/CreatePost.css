textarea {
  overflow: auto;
  height: auto;
}

label {
  font-weight: 600;
  max-width: 100%;
  display: block;
  margin: 1rem 0 0.5rem;
}

form {
  background: white;
  padding: 30px;
  border-radius: 5px;
  box-shadow: 3px 3px 10px 0px #aaa;
}

.validate-fail {
  border: 1px solid red;
}

.message-fail {
  color: red;
}
.message-success {
  color: green;
}
