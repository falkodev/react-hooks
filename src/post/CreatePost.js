import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CreatePost = props => {
  const initialState = {
    content: '',
    tags: '',
    author: '',
  }
  const [post, setPost] = useState(initialState)
  const [message, setMessage] = useState('')

  const handleInputChange = event => {
    const { name, value } = event.target
    setMessage('')
    document.getElementById('content').classList.remove('validate-fail')
    setPost({ ...post, [name]: value })
  }

  const handleSubmit = event => {
    event.preventDefault()
    if (!post.content) {
      setMessage('Missing field content.')
      document.getElementById('message').classList.add('message-fail')
      document.getElementById('content').classList.add('validate-fail')
      return
    }

    props.addPost(post, props.setPosts)
    setPost(initialState)
    setMessage('Post added.')
    document.getElementById('message').classList.add('message-success')
  }

  return (
    <form onSubmit={handleSubmit}>
      <div className="Post-Content">
        <label htmlFor="content">Content</label>
        <textarea
          name="content"
          id="content"
          value={post.content}
          onChange={handleInputChange}
        />
      </div>
      <div className="Post-Tags">
        <label htmlFor="tags">Tags</label>
        <input
          type="text"
          name="tags"
          id="tags"
          value={post.tags}
          onChange={handleInputChange}
        />
      </div>
      <div className="Post-Author">
        <label htmlFor="author">Author</label>
        <input
          type="text"
          name="author"
          id="author"
          value={post.author}
          onChange={handleInputChange}
        />
      </div>
      <br />
      <input type="submit" data-testid="submit" value="Submit" />
      <label id="message">{message}</label>
    </form>
  )
}

CreatePost.propTypes = {
  content: PropTypes.string,
  tags: PropTypes.array,
  author: PropTypes.string,
}

export default CreatePost
