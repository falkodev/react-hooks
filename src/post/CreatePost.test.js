import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import CreatePost from './CreatePost'

it('adds a post', () => {
  const posts = []
  const addPost = post => {
    post.id = posts.length + 1
    posts.push(post)
  }

  const { getByTestId, getByLabelText } = render(
    <CreatePost addPost={addPost} />,
  )
  const input = getByLabelText('Content')
  fireEvent.change(input, { target: { value: 'test' } })
  fireEvent.click(getByTestId('submit'))

  expect(posts.length).toBe(1)
})
