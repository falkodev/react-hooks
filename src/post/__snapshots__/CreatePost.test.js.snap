// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a post 1`] = `
<form>
  <div
    class="Post-Content"
  >
    <label
      for="content"
    >
      Content
    </label>
    <textarea
      id="content"
      name="content"
    />
  </div>
  <div
    class="Post-Tags"
  >
    <label
      for="tags"
    >
      Tags
    </label>
    <input
      id="tags"
      name="tags"
      type="text"
      value=""
    />
  </div>
  <div
    class="Post-Author"
  >
    <label
      for="author"
    >
      Author
    </label>
    <input
      id="author"
      name="author"
      type="text"
      value=""
    />
  </div>
  <br />
  <input
    type="submit"
    value="Submit"
  />
  <label
    id="message"
  />
</form>
`;
